drivers: display: st7735r: added configuration for color inverting

soburi opened the pull request:
Add 'invert-color' configuration to st7735r dts-node.

Issue INVON(21h) command on initializing if invert-color was enabled.
As a result of this command, the display color is inverted.
Otherwise, INVOFF(20h) will be issued.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

#44796 is depends to this patch.

jfischer-no commented on dts/bindings/display/sitronix,st7735r.yaml:
```diff
@@ -115,3 +115,7 @@ properties:
+
+    invert-color:
+      type: boolean
+      description: Display color inversion
```
Since you are touching it, can you please fix description from `invctr` above to "Display Inversion mode"? If so please do it in separate commit.

The name and description of property you would like to add should probably be:

```
    inversion-on:
      type: boolean
      description: Enable Display Inversion
```

soburi replied:
Splitting commit and Fix it.
Changing variable name as well.

jfischer-no replied:
>Since you are touching it, can you please fix description from `invctr` above to "Display Inversion mode"? If so please do it in separate commit.

You do not need to split your commit. I asked if you could fix the description of existing `invctr` property because it is a bit confusing now, if so, it should be done in separate commit.

soburi replied:
I got it.
I updated description of 'invctr' with separeted commit with  adding short summary of datasheet's description.

soburi committed:
drivers: display: st7735r: added inversion-on property

Add 'inversion-on' property to st7735r.
Issue INVON(21h) command on initializing if inversion-on was enabled.
As a result of this command, the display color is inverted.
Otherwise, INVOFF(20h) will be issued.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

mbolivar-nordic reviewed (commented):
LGTM, @jfischer-no please take another look

soburi committed:
dts: bindings: display: st7735r: add more description for invctr

Add more description for 'invctr' property.
Add information about what this property inverts.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

jfischer-no commented on dts/bindings/display/sitronix,st7735r.yaml:
```diff
@@ -117,3 +117,9 @@ properties:
+
+    inversion-on:
+      type: boolean
+      description: Enable Display Inversion
```
Not sure about style here, we have different in the tree, but @mbolivar-nordic will complain if necessary.
```
      description: |

        Enable Display Inversion
        Make a drawing with the inverted color of the frame memory.
```

mbolivar-nordic replied:
@jfischer-no is right; this needs to be done in a different way. Like this, please:

```
description: |
    Enable Display Inversion
    Make a drawing with the inverted color of the frame memory.
```

The same change should be made for the `invctr` property also.

Please see https://docs.zephyrproject.org/latest/build/dts/bindings.html#general-rules for an explanation about why this is needed.

soburi replied:
@jfischer-no , @mbolivar-nordic 
Thanks for reviewing.
I Fixed it.
